File: programmers/43165/43165.py
def solution(numbers, target):
    answer = 0
    def dfs(start,idx) :
        nonlocal answer
        if idx == len(numbers) :
            if start == target :
                answer += 1
            return
        dfs(start-numbers[idx],idx+1)
        dfs(start+numbers[idx],idx+1) 
    dfs(0,0)
    return answer

File: C++/programmers/43165/43165.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;


int solution(vector<int> numbers, int target) {
    int answer = 0;
    stack<pair<int,int>> s;
    s.push({0,0});
    while(!s.empty()) {
        auto cur = s.top();
        s.pop();
        int idx = cur.first;
        int sum = cur.second;
        if(idx == numbers.size()) {
            if(sum == target) answer++;
            continue;
        }
        s.push({idx + 1, sum+numbers[idx]});
        s.push({idx + 1, sum-numbers[idx]});
    }
    return answer;
}